napari/utils/_magicgui.py

```diff
@@ -198,6 +219,110 @@ def add_layer_data_tuples_to_viewer(gui, result, return_type):
             viewer._add_layer_from_data(*layer_datum)
 
 
+_FUTURES: Set[Future] = set()
+
+
+def add_worker_data(
+    gui, worker: 'FunctionWorker', return_type, _from_tuple=True
+):
+    """Handle a thread_worker object returned from a magicgui widget.
+
+    This allows someone annotate their magicgui with a return type of
+    `FunctionWorker[...]`, create a napari thread worker (e.g. with the
+    @thread_worker decorator), then simply return the worker.  We will hook up
+    the `returned` signal to the machinery to add the result of the
+    long-running function to the viewer.
+
+    Parameters
+    ----------
+    gui : MagicGui
```

Comment:
I wonder if there is a better kwarg here than `gui` (which makes me think of whole viewer). I guess is is a magicgui widget? Not critical to change, but just flagging in case anyone has another idea

Reply:
Maybe it's ok - I'm looking at the doc string for `add_future_data` too and it does make sense

Reply:
yeah, I agree actually... I think the docstrings of that whole module need some updating